tests/phpunit/integration/Modules/Thank_With_Google/Web_TagTest.php

```diff
@@ -0,0 +1,149 @@
+		$this->web_tag->register();
+
+		$this->create_post_and_go_to_it();
+
+		$output = $this->capture_output( 'the_content' );
+
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( '<button twg-button', $output );
+	}
+
+	public function test_content_placeholder_not_inserted_on_dynamic_button_placement() {
+		remove_all_filters( 'the_content' );
+
+		$this->web_tag->set_button_placement( 'dynamic_low' );
+		$this->web_tag->register();
+
+		$this->create_post_and_go_to_it();
+
+		$output = $this->capture_output( 'the_content' );
+
+		$this->assertStringNotContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringNotContainsString( '<button twg-button', $output );
+	}
+
+	public function test_content_placeholder_inserted_static_below_content() {
+		remove_all_filters( 'the_content' );
+
+		$this->web_tag->set_button_placement( 'static_below-content' );
+		$this->web_tag->register();
+
+		$this->create_post_and_go_to_it();
+
+		$output  = $this->capture_output( 'the_content' );
+		$content = get_the_content();
+
+		$this->assertStringStartsWith( $content, $output );
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( '<button twg-button', $output );
+	}
+
+	public function test_content_placeholder_inserted_static_above_content() {
+		remove_all_filters( 'the_content' );
+
+		$this->web_tag->set_button_placement( 'static_above-content' );
+		$this->web_tag->register();
+
+		$this->create_post_and_go_to_it();
+
+		$output  = $this->capture_output( 'the_content' );
+		$content = get_the_content();
+
+		$this->assertStringEndsWith( $content, $output );
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( '<button twg-button', $output );
+	}
+
+	public function test_content_placeholder_inserted_static_below_first_paragraph() {
+		remove_all_filters( 'the_content' );
+
+		$this->web_tag->set_button_placement( 'static_below-first-paragraph' );
+		$this->web_tag->register();
+
+		$this->create_post_and_go_to_it();
+
+		$output = $this->capture_output( 'the_content' );
+
+		$this->assertStringStartsWith( '<p>Hello World</p>', $output );
+		$this->assertStringEndsWith( '<p>Goodbye World</p>', $output );
+		$this->assertStringContainsString( 'Thank with Google snippet added by Site Kit', $output );
+		$this->assertStringContainsString( '<button twg-button', $output );
+	}
+
+	private function create_post_and_go_to_it() {
+		$post_ID = $this->factory()->post->create(
+			array(
+				'post_content' => '<p>Hello World</p><p>Goodbye World</p>',
+			)
+		);
+		$this->go_to( get_permalink( $post_ID ) );
+	}
+
+	private function capture_output( $callback ) {
+		ob_start();
+		$callback();
+		$output = ob_get_clean();
+		return $output;
+	}
```

Comment:
This method is not needed. There is the `get_echo` function that does the same: https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/includes/utils.php#L351-L355